Reference/using-ioc.md

```diff
@@ -141,3 +141,103 @@ If you don't register assemblies that contain controllers you may end up with YS
+
+##Unity Example
+
+Install the Unity, Unity.Mvc and Unity.AspNet.WebApi packages.
+Remove the Mvc and Api specific files created, we will be merging those files shortly.
+Leave the UnityConfig file as is.
+
+Now create the following files to configure Unity correctly for an Umbraco site.
+
+	class UnityEvents : IApplicationEventHandler
+	{
+	    public void OnApplicationStarted(
+		UmbracoApplicationBase httpApplication,
+		ApplicationContext applicationContext
+	    )
+	    {
+		var container = UnityConfig.GetConfiguredContainer();
+
+		// If you intend to use DI inside filter attributes
+		FilterProviders.Providers.Remove(FilterProviders.Providers.OfType<FilterAttributeFilterProvider>().First());
+		FilterProviders.Providers.Add(new UnityFilterAttributeFilterProvider(container));
+		// Web API
+		GlobalConfiguration.Configuration.DependencyResolver 
+			= new Microsoft.Practices.Unity.WebApi.UnityDependencyResolver(container);
+		// MVC
+		DependencyResolver.SetResolver(new Microsoft.Practices.Unity.Mvc.UnityDependencyResolver(container));
+
+		// This will automatically scan an assembly for classes fitting this convention:
+		// Interface name: IMyClass
+		// Class name: MyClass
+		// and automatically register those types for you.
+		// If you do decide to use this feature, make sure it is the first one called.
+		// Conflicting registrations that follow will override previous ones.
+		container.RegisterTypes(
+		    AllClasses.FromAssemblies(typeof(UnityEvents).Assembly),
+		    WithMappings.FromMatchingInterface,
+		    WithName.Default
+		);
+
+		// The UmbracoContext must be registered so that the umbraco backoffice controllers 
+		// can be successfully resolved
+		container.RegisterType<UmbracoContext>(
+			new PerRequestLifetimeManager(), 
+			new InjectionFactory(c => UmbracoContext.Current)
+		);
+
+		// Instantiate the HealthCheckController using it's default parameterless constructor
+		container.RegisterType<HealthCheckController>(new InjectionConstructor());
+		
+		// Some common examples
+		
+		// The commonly used Umbraco ApplicationContext
+		container.RegisterType<ApplicationContext>(
+		    new ContainerControlledLifetimeManager(), 
+		    new InjectionFactory(c => ApplicationContext.Current)
+		);
+
+		// An HttpContext abstraction that you may find useful
```

Comment:
I'd rather keep this out of the docs as it's not necessary for Umbraco to operate and I don't want to add any confusion